File: clients/c#/ownPush/ConnectionObject.cs
namespace ownPush
{
    enum Purpose
    {
        ERROR = -1,
        REQUEST = 0,
        CHALLENGE = 1,
        LOGIN = 2,
        PUSH = 3,
        RESET = 4
    }

    class ConnectionObject
    {
        public ConnectionObject(Purpose pur, string d)
        {
            purpose = pur;
            data = d;
        }

        public Purpose purpose;
        public string data;

        public override string ToString()
        {
            return "Purpose: " + purpose + " | " + data;
        }
    }
}

File: clients/android/app/src/main/java/net/damaex/OwnPush/ConnectionObject.java
package net.damaex.OwnPush;

class ConnectionObject {
    public enum Purpose {
        ERROR(-1),
        REQUEST(0),
        CHALLENGE(1),
        LOGIN(2),
        PUSH(3),
        RESET(4);

        private final int type;

        Purpose(int type) {
            this.type = type;
        }

        public int getType() {
            return this.type;
        }
    }

    ConnectionObject() {
        this.purpose = Purpose.ERROR;
        this.data = "";
    }

    ConnectionObject(Purpose purpose, String data) {
        this.purpose = purpose;
        this.data = data;
    }

    Purpose purpose;
    String data;
}
